src/main/scala/kamon/newrelic/metrics/NewRelicMetricsReporter.scala

```diff
@@ -0,0 +1,72 @@
+/*
+ *  Copyright 2019 New Relic Corporation. All rights reserved.
+ *  SPDX-License-Identifier: Apache-2.0
+ */
+
+package kamon.newrelic.metrics
+
+import com.newrelic.telemetry.SimpleMetricBatchSender
+import com.newrelic.telemetry.metrics.{MetricBatch, MetricBatchSender}
+import com.typesafe.config.Config
+import kamon.Kamon
+import kamon.metric.PeriodSnapshot
+import kamon.module.{MetricReporter, Module, ModuleFactory}
+import kamon.newrelic.AttributeBuddy.buildCommonAttributes
+import org.slf4j.LoggerFactory
+
+import scala.jdk.CollectionConverters._
+
+class NewRelicMetricsReporter(senderBuilder: () => MetricBatchSender = () => NewRelicMetricsReporter.buildSender()) extends MetricReporter {
+
+  private val logger = LoggerFactory.getLogger(classOf[NewRelicMetricsReporter])
+  @volatile private var commonAttributes = buildCommonAttributes(Kamon.config())
+  @volatile private var sender: MetricBatchSender = senderBuilder()
+
+  override def reportPeriodSnapshot(snapshot: PeriodSnapshot) = {
+    logger.warn("NewRelicMetricReporter reportPeriodSnapshot...")
```

Comment:
This probably should be debug instead of warn.

Reply:
yes. will change.

Reply:
done